Chatspace group メンバー追加機能

teru0225 opened the pull request:
WHY
グループ作成機能を作る上で、必要不可欠であるため

WHAT
メンバー追加機能の搭載
・groups_controllerの編集
・views/groups/new.html.hamlの編集

https://gyazo.com/d90edea57589d5b321e168d872c5b1d3
https://gyazo.com/5d9a60f6c7c9c2589fc2b257d0cc3b79

teru0225 committed:
Edited groups_controller.rb

Added
- @group = Group.new
- @group.users << current_user
Newメソッドにこれらを入力することで、グループ作成時に、現在ログインしているユーザーが、groupモデルのusersカラムに入る。
-  @group = Group.new(group_params)
- if @group.save
      redirect_to root_path, notice: "グループを作成しました。"
    else
      render action: :new
    end
- private
  def group_params
 params.require(:group).permit(:name)
end
ストロングパラメーターで、groupの値だけを取得するとともに、group内のnameカラムのみを取得できるようにている。
また、グループの作成をできたときには、rootであるmessage#indexに戻るようにし、その際にはflashメッセージが出るようにした。
グループの作成ができなかったときには、renderアクションで、newアクションに戻るようにした

teru0225 committed:
Edited new.html.haml

Converted form tags
- もし、グループが作成できなかったときには、エラーが出るようにしました。

kuroko918 commented on app/views/groups/new.html.haml:
```diff
@@ -1,16 +1,17 @@
-      %h2
-        1件のエラーが発生しました。
+  = form_for @group do |f|
+    - if @group.nil?
```
@groupがnilになることはないかと思うのですが…？

teru0225 replied:
@group.errors.any?
に書き換えました。

kuroko918 replied:
errorを持つこと自体ありますか？

teru0225 replied:
errorを持つこと自体はあると思います。
例えば、この場合nameを記入しない状態だと、エラーが出るというのが私の認識です。

kuroko918 commented on app/views/groups/new.html.haml:
```diff
@@ -28,6 +29,7 @@
         %label.chat-group-form__label{:for => "chat_group_チャットメンバー"} チャットメンバー
       .chat-group-form__field--right
         / グループ作成機能の追加時はここにcollection_check_boxesの記述を入れてください
+        = f.collection_check_boxes :user_id, User.all, :id, :name
```
こちらの記述が気になるため、一度画面のスクショを下さい！

teru0225 replied:
user_idsに変えました。
ご指摘いただいた２点を直す上で、下記を参考にしました。
レビューありがとうございます。
https://techacademy.jp/magazine/10024

teru0225 committed:
Edited routes.rb

In order to check if groups#new works or not

teru0225 committed:
Revert "Edited routes.rb"

This reverts commit d317b9f41632bc346919b5b29ffb47c7c66ced24.

teru0225 committed:
Edited routes.rb

In order to check if groups#new works or not

teru0225 committed:
Edited groups/new.html.haml

if @group.nil?
=>
if @group.errors.any?
Groupがnilになることはないため
「もし@group変数にエラーがある場合」という条件式に変えた

%li エラーです
=>
 - @group.errors.full_messages.each do |message|
            %li= message
エラー文がエラーが発生した際に出るようにした

        = f.collection_check_boxes :user_id, User.all, :id, :name
=>
        = f.collection_check_boxes :user_ids, User.all, :id, :name
user_idでは、認識してくれないため。
複数形にする必要がある。

teru0225 committed:
Edited new.html.haml

kuroko918 commented on config/routes.rb:
```diff
@@ -1,7 +1,7 @@
 Rails.application.routes.draw do
   devise_for :users
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
```
不要なコメントは消しておきましょう！

teru0225 replied:
ご指摘いただいたコメントアウトを消しました！

kuroko918 commented on app/views/groups/new.html.haml:
```diff
@@ -1,16 +1,18 @@
+    - if @group.errors.any?
+      .chat-group-form__errors
+        %h2
+          1件のエラーが発生しました。
```
エラーを表示させるのでしたら、件数も常に1件ではなく、errorの件数をきちんと表示させてげましょう！

teru0225 replied:
countメソッドを使うことで、エラーの件数を表示できるようにしました。

kuroko918 commented on config/routes.rb:
```diff
@@ -1,7 +1,7 @@
-  root 'messages#index'
+  root 'groups#new'
   resources :users, only: [:index, :edit, :update]
   resources :groups, only: [:new, :create, :edit, :update] do
```
rootにnewアクションを指定した場合は、groupsのonlyに指定されているnewは消しておきましょう！

teru0225 replied:
ご指摘ありがとうございます。
結局、rootはmessages#indexに戻しました。

kuroko918 commented:
見た目のスクショを下さい！

teru0225 committed:
edited groups/new.html.haml

          1件のエラーが発生しました。
=>
          = "#{@group.errors.full_messages.count}件のエラーが発生しました。"
エラーの件数を件数によって表示が変わるようにした。

teru0225 committed:
Edited routes.rb

  root 'groups#new'
=>
  root 'messages#index'

teru0225 commented:
> 見た目のスクショを下さい！

プルリクのWHATの下に貼っておきました。
確認よろしくお願いします。

kuroko918 commented:
いいかと思います！
LGTMです！

teru0225 commented:
> いいかと思います！
> LGTMです！

レビューありがとうございました！

teru0225 committed:
Edited groups/new.html.haml

クラス名の記入ミスがあったため、そこを直しました。
